staging/src/k8s.io/controller-manager/pkg/clientbuilder/client_builder_dynamic.go

```diff
@@ -235,7 +235,9 @@ func (ts *tokenSourceImpl) Token() (*oauth2.Token, error) {
 
 func constructClient(saNamespace, saName string, config *restclient.Config) restclient.Config {
 	username := apiserverserviceaccount.MakeUsername(saNamespace, saName)
-	ret := *restclient.AnonymousClientConfig(config)
```

Comment:
Hmm... we called this to make sure it doesn't hold any identifiable data. Alternative are:
1. Add 2nd method which is similar to `AnonymousClientConfig` but leaves transports
2. Use `AnonymousClientConfig` but copy transport.

Reply:
> Hmm... we called this to make sure it doesn't hold any identifiable data.

yes so this is what we already have for KCM no need to call `AnonymousClientConfig` for each client
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/controllermanager.go#L223

Reply:
Fair.